src/macro_api.c

```diff
@@ -0,0 +1,22 @@
+#include "reactor-uc/logging.h"
+#include "reactor-uc/action.h"
+#include "reactor-uc/scheduler.h"
+#include "reactor-uc/reactor.h"
+#include "reactor-uc/environment.h"
+#include <assert.h>
+#include <string.h>
+
+/**
+ * @brief this is a helper function for the lf_schedule macro with a value
+ **/
+lf_ret_t lf_schedule_with_value(Action* action, interval_t offset, const void* val) {
+  lf_ret_t ret = action->schedule(action, offset, val);
+  if (ret == LF_FATAL) {
+    LF_ERR(TRIG, "Scheduling a value, for an void action!");
+    Scheduler* sched = action->super.parent->env->scheduler;
+    sched->do_shutdown(sched, sched->current_tag(sched));
+    throw("Tried to schedule a value onto an action without a type!");
```

Comment:
Are two error messages needed here?  I'm not sure what `throw` does.

Reply:
`throw` on many embedded platforms halts the execution and prints this message. I think this is what we want, because scheduling a void-action with a value, should be a hard crash.

Reply:
I cannot find any reference to `throw` in any C standard. Can you provide a pointer?